devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy

```diff
@@ -90,46 +163,55 @@ def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
 	log.debug "Basic ${cmd}" + (ep ? " from endpoint $ep" : "")
 	def value = cmd.value ? "on" : "off"
-	ep ? changeSwitch(ep, value) : []
+
+	if (isZoozZenStripV2()) {
+		// device sends also reports without any endpoint specified, therefore all endpoins must be querried
+		// sometimes it also reports 0.0 Wattage only until it's queried for it, then it starts reporting real values
+		ep ? [changeSwitch(ep, value), response(secureEncap(zwave.meterV3.meterGet(scale: 0), ep))] : [response(refreshBasicGetAll())]
+	} else {
+		ep ? changeSwitch(ep, value) : []
+	}
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd, ep = null) {
 	log.debug "Binary ${cmd}" + (ep ? " from endpoint $ep" : "")
 	def value = cmd.value ? "on" : "off"
-	ep ? changeSwitch(ep, value) : []
+
+	if (isZoozZenStripV2()) {
+		// device sends also reports without any endpoint specified, therefore all endpoins must be querried
+		// sometimes it also reports 0.0 Wattage only until it's queried for it, then it starts reporting real values
+		ep ? [changeSwitch(ep, value), response(secureEncap(zwave.meterV3.meterGet(scale: 0), ep))] : [response(refreshBasicGetAll())]
+	} else {
+		ep ? changeSwitch(ep, value) : []
+	}
 }
 
 private changeSwitch(endpoint, value) {
-	def result = []
 	if (endpoint == 1) {
-		result += createEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+		createEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
 	} else {
 		String childDni = "${device.deviceNetworkId}:$endpoint"
 		def child = childDevices.find { it.deviceNetworkId == childDni }
 		child?.sendEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
 	}
-	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def event = createEvent([isStateChange:  false, descriptionText: "Wattage change has been detected. Refreshing each endpoint"])
+	isAeotec() ? [event, response(refreshAll())] : event
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep) {
```

Comment:
while you're refactoring, can you combine this and the other meter report handler and just set the default value for `ep`?

Reply:
sure, fixed